core/domain/user_services.py

```diff
@@ -2023,7 +2021,9 @@ def log_username_change(committer_id, old_username, new_username):
         new_username: str. The new username that the current one is being
             changed to.
     """
-    model_id = '%s.%s' % (committer_id, int(math.floor(time.time())))
+    time_of_query = (datetime.datetime.utcnow() - datetime.datetime(
```

Comment:
What's wrong with the old method?

If the issue is a flaky backend test, I would expect the change to instead be to mock time.time() and have it return a specific value for the context of the test...

If you actually need to make a change to this method, then I would be fine with using utils.get_current_time_in_millisecs() since that's sort of our "standard". We should not be doing random time operations throughout the codebase.

Reply:
Hi, do we have any `UsernameChangeAuditModel` already in the datastore? Depending on that I will pick the approach.

Reply:
No, not yet.

Reply:
Done.